lib/index.d.ts

```diff
@@ -33,7 +33,7 @@ interface UploadOptions {
   onSuccess?: (() => void) | null
   onError?: ((error: Error | DetailedError) => void) | null
   onShouldRetry?:
-    | ((error: Error | DetailedError, retryAttempt: number, options: UploadOptions) => boolean)
+    | ((error: DetailedError, retryAttempt: number, options: UploadOptions) => boolean)
```

Comment:
Please keep `Error` here as the type is not always DetailedError: https://github.com/tus/tus-js-client/pull/287#issuecomment-945199333

Reply:
But that's not revevant here.  In `shouldRetry` calls `onShouldRetry` if `err.originalRequest != null` which means that it can't just be an error.

Reply:
Good point, then let's change it 👍